sympy/polys/domains/domain.py

```diff
@@ -14,7 +14,7 @@
 from sympy.utilities import default_sort_key, public
 
 @public
-class Domain(object):
+class Domain(Basic):
```

Comment:
I am not sure that this would be a good idea. `Basic` objects have a great number of attributes and methods that are not meaningful for domains. If it is necessary to have `Basic` objects, I would consider defining some kind of a wrapper analogous to `Tuple` or `Dict`.

Reply:
`Tuple` subclasses from `Basic` and I am guessing it also does not make use of a large number of `Basic` attributes? I understand where @sylee957 is going with this and it could be a big help in matrices if we could have a proper sympifiable `DomainMatrix` with sympifiable domain. What about overloading the attributes and methods which make no sense for `Domain` to raise?

Reply:
Do you mean that if `polys.domains` should not be subclassed from Basic, should an another object with similar names should be introduced for sympy frontend usage?

I'm actually thinking of building a new module for `Ring`, `Field` that are analogous to the `Set` module, if it should be like that.

Reply:
I was thinking of a wrapper class, say `Ring`, that could be a subclass of `Basic` and have an attribute `ring` that could be an instance of `Domain`, or even something more general like a quaternion or matrix algebra.